server/dive_server/utils.py

```diff
@@ -3,16 +3,38 @@
 import os
 from datetime import datetime
 from pathlib import Path
-from typing import Dict
+from typing import Dict, Iterable, Optional
 
+from girder.exceptions import RestException
 from girder.models.file import File
 from girder.models.folder import Folder
 from girder.models.item import Item
 from girder.models.upload import Upload
+from girder.models.user import User
 
 from dive_server.serializers import viame
 
 
+def all_detections_items(folder: Folder) -> Iterable[Item]:
+    """caller is responsible for verifying access permissions"""
+    return Item().find({"meta.detection": str(folder['_id'])}).sort([("created", -1)])
+
+
+def detections_item(folder: Folder, strict=False) -> Optional[Item]:
+    all_items = all_detections_items(folder)
+    detection = all_items[0] if all_items else None
```

Comment:
`all_detection_items` can return an empty Cursor object which is still true.  This is the case if you upload media with no source csv detections.  This could also be the case if the detections were removed from the girder folder manually so there was no return value.
something like ` all_items[0] if all_items.count() > 0 else None` may work?